figures/scripts/fig3-chromothripsis-barplot.R

```diff
@@ -0,0 +1,120 @@
+if (!dir.exists(output_dir)) {
+  dir.create(output_dir, recursive = TRUE)
+}
+
+# Data directory
+data_dir <- file.path(root_dir, "data")
+
+# Analysis directory
+analyses_dir <- file.path(root_dir, "analyses")
+
+# Chromothripsis results directory
+chromothripsis_dir <- file.path(analyses_dir, "chromothripsis", "results")
+
+#### Files ---------------------------------------------------------------------
+
+# Histologies file
+histologies_file <- file.path(data_dir, "pbta-histologies.tsv")
+
+# Results file with chromothripsis info for each sample
+chromo_file <- file.path(chromothripsis_dir,
+                         "chromothripsis_summary_per_sample.txt")
+
+# Most recent palette for cancer group
+palette_file <- file.path(figures_dir,
+                          "palettes",
+                          "broad_histology_cancer_group_palette.tsv")
+
+#### Read in data --------------------------------------------------------------
+
+# Read in clinical data
+histologies_df <- read_tsv(histologies_file, guess_max = 10000)
+
+# Read in chromothripsis per sample
+chromo_per_sample_df <- read_tsv(chromo_file)
+
+# Read in most recent cancer group palette
+palette_df <- read_tsv(palette_file)
+
+#### Prep ----------------------------------------------------------------------
+
+# We'll only be using the cancer group palette and we need a named vector of
+# hexcodes
+cancer_group_col_df <- palette_df %>%
+  select(cancer_group, cancer_group_hex) %>%
+  distinct()
+
+# Can't do pull with names with our versions
+cancer_group_palette <- cancer_group_col_df$cancer_group_hex
+names(cancer_group_palette) <- cancer_group_col_df$cancer_group
+
+# Need a data frame that contains the identifiers + cancer group
+cancer_group_df <- histologies_df %>%
+  select(Kids_First_Biospecimen_ID, cancer_group)
+
+# Add cancer group and palette information t
+chromo_per_sample_df <- chromo_per_sample_df %>%
+  left_join(cancer_group_df)
+
+### Repeat all of these steps for cancer_group instead of display_group
+chromoth_cancer_group_df <- chromo_per_sample_df %>%
+  count(any_regions_logical, cancer_group) %>%
+  tidyr::spread(key = any_regions_logical, value = n, fill=0) %>%
+  group_by(cancer_group) %>%
+  mutate(cancer_group_size = sum(`TRUE`, `FALSE`)) %>%
+  mutate(prop = `TRUE` / cancer_group_size) %>%
+  mutate(labels = paste0(`TRUE`, " / ", cancer_group_size)) %>%
+  ungroup(cancer_group) %>%
+  filter(cancer_group_size >= 3 & !is.na(cancer_group)) # Only keep groups with >=3 tumors
+
+# Reorder cancer_group based on proportion and remove any cancer groups with
+# no instances of chromothripsis
+chromoth_cancer_group_df <- chromoth_cancer_group_df  %>%
+  mutate(cancer_group = fct_reorder(cancer_group, prop))
+
+#### Plotting ------------------------------------------------------------------
+
+# Set ggplot2 theme
+theme_set(theme_pubr())
+
+# Save ggplot2 options
+plot_options <- list(
+  ylim(c(0, 1)),
+  xlab(NULL),
+  ylab("Proportion of Tumors"),
```

Comment:
```suggestion
  ylab("Proportion of Tumors with Chromothripsis Events"),
```
I don't think it would hurt to be more specific about what the proportion of tumors is actually referring to, even though it will probably be in the legend, I think it makes it easier on readers and reviewers.